### PR TITLE
crimson: check -ENAMETOOLONG for Name, Locator, NameSpace

### PR DESCRIPTION
**Pull request by liu-chunmei**
Signed-off-by: chunmei-liu <chunmei.liu@intel.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/crimson/osd/osd_operations/client_request.cc [hunk 1]
```diff
@@ -209,6 +209,19 @@ ClientRequest::process_op(Ref<PG> &pg)
     return seastar::make_ready_future<seq_mode_t>(seq_mode_t::OUT_OF_ORDER);
   }));
 }
+auto ClientRequest::reply_op_error(Ref<PG>& pg, int err)
```
Comment (athanatos) [inline src/crimson/osd/osd_operations/client_request.cc:212]: Add a newline between the prior method and this one.

File: src/crimson/osd/osd_operations/client_request.cc [hunk 2]
```diff
@@ -220,15 +233,24 @@ ClientRequest::do_process(Ref<PG>& pg, crimson::osd::ObjectContextRef obc)
...
-      auto reply = crimson::make_message<MOSDOpReply>(
-	m.get(), -EAGAIN, pg->get_osdmap_epoch(),
-	m->get_flags() & (CEPH_OSD_FLAG_ACK|CEPH_OSD_FLAG_ONDISK),
-	!m->has_flag(CEPH_OSD_FLAG_RETURNVEC));
-      return conn->send(std::move(reply)).then([] {
-        return seastar::make_ready_future<seq_mode_t>(seq_mode_t::OUT_OF_ORDER);
-      });
+      return reply_op_error(pg, -EAGAIN);
     }
   }
```
Comment (athanatos) [inline src/crimson/osd/osd_operations/client_request.cc:238]: } else if { for all of these

**Review by athanatos — APPROVED** [review]
LGTM other than those nits.

File: src/crimson/osd/osd_operations/client_request.cc [hunk 3]
```diff
@@ -220,15 +233,24 @@ ClientRequest::do_process(Ref<PG>& pg, crimson::osd::ObjectContextRef obc)
...
+      > crimson::common::local_conf()->osd_max_object_name_len) {
+    return reply_op_error(pg, -ENAMETOOLONG);
+  }
+  if (m->get_hobj().nspace.size()
+      > crimson::common::local_conf()->osd_max_object_namespace_len) {
+    return reply_op_error(pg, -ENAMETOOLONG);
+  }
+  if (m->get_hobj().oid.name.empty()) {
+    return reply_op_error(pg, -EINVAL);
+  }
```
Comment (rzarzynski) [inline src/crimson/osd/osd_operations/client_request.cc:253]: How about dissecting these sanity checks into a dedicated method?

Comment (liu-chunmei) [reply src/crimson/osd/osd_operations/client_request.cc:252]: I keep it as it, it hardly handle the continuation logic.

**Pull request merged**